pkg/server/main_context_test.go

```
@@ -67,3 +69,50 @@ func TestRunServerContextReturnsCancellation(t *testing.T) {
+	defer func() {
+		if closeErr := handler.Close(); closeErr != nil {
+			t.Errorf("close command handler: %v", closeErr)
+		}
+	}()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	shutdown, err := startInternalBrokerListener(ctx, cfg, handler)
+	if err != nil {
+		t.Fatal(err)
+	}
+	conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", port))
```

golangci-lint (2.12.2)

[error] net.Dial must not be called. use (*net.Dialer).DialContext

(noctx)